### PR TITLE
[frontend] detail view, part 2

### PR DESCRIPTION
**Pull request by dingens**
This completes the detail view by adding information about flavors. (There's stlll a few `#TODO` notes -- I'd like to merge this nonetheless for now so that all data is finally visible on the site. I'd fix those later.)

Also, I've updated the scheme list on the start page. It now a) shows some basic info (category, description, nist round), b) links to the scheme's detailed view. This includes some code restructuring, mainly using the (now global) `SCHEME_TYPES` dict that assembles some data we need everywhere (like mapping `ct_sig` to either `ct` or `sig`).

### CODE REVIEW

File: frontend/src/components/views.js [hunk 1]
```diff
@@ -848,24 +872,25 @@ class SchemeDetail extends React.Component {
...
-                                const f_links = queryAll(db, "SELECT * FROM flavor_link WHERE flavor_id=?", [f.id]);
-                                const f_sources = queryAll(db, "SELECT * FROM flavor_source WHERE flavor_id=?", [f.id]);
+                                const f_paramsets = queryAll(db, "SELECT * FROM paramset WHERE flavor_id=? ORDER BY security_level_nist_category ASC, security_level_quantum ASC", [f.id]);
+                                const f_implementations = queryAll(db, "SELECT name FROM implementation WHERE flavor_id=? ORDER BY type DESC", [f.id]);
 
                                 return [
                                     <ListItem key={"flavor-" + f.id + "-head"} style={{ display: "block" }}>
-                                        <Typography component="h3" variant="h5">{f.name}</Typography>
+                                        <Typography component="h3" variant="h5">
+                                            <Link href={"?_=" + s.type + "/" + s.id + "/" + f.id}>{f.name}</Link>
```
Comment (benj-zen) [inline frontend/src/components/views.js:881]: This should not use `href` but instead `component={RouterLink}` and `to` as for instance in line 247.

File: frontend/src/components/views.js [hunk 2]
```diff
@@ -904,13 +924,240 @@ class SchemeDetail extends React.Component {
...
+        let TypeIcon = SCHEME_TYPES[s.type].icon;
+        return [
+            <Container maxWidth="md">
+                <Paper>
+                    <Box>
+                        <List>
+                            <ListItem key="scheme-head" alignItems="flex-start">
+                                <ListItemText>
+                                    <Typography variant="h6">
+                                        <Link href={ "?_=" + s.type + "/" + s.id }>
```
Comment (benj-zen) [inline frontend/src/components/views.js:946]: See comment above

File: frontend/src/components/views.js [hunk 3]
```diff
@@ -904,13 +924,240 @@ class SchemeDetail extends React.Component {
...
+                                <PropItem k={ "i-" + i.id + "-sources" } title="Sources" icon={ SourceIcon }>
+                                    { i_sources.map(s => <div>{ linkify(s.url) }</div>) }
+                                </PropItem>,
+
+                            ] } ) }
+
+                            <ListItem key="benchmarks">
+                                <ListItemText>
+                                    <Typography component="h3" variant="h4">Benchmarks</Typography>
+                                    <div><Link href={"raw_sql?query=SELECT p.name AS 'Parameter Set'%2C i.name AS Implementation%2C b.platform AS Platform%2C b.comment AS '🛈'%2C b.timings_unit%2C b.timings_gen%2C b.timings_enc_sign%2C b.timings_dec_vrfy%2C b.timings_comment AS '🛈'%2C b.memory_requirements_gen%2C b.memory_requirements_enc_sign%2C b.memory_requirements_dec_vrfy%2C b.memory_requirements_comment AS '🛈' FROM benchmark b%2C paramset p%2C implementation i WHERE p.id%3Db.paramset_id AND i.id%3Db.implementation_id AND p.flavor_id%3D" + f.id}>Show all benchmarks for this flavor</Link></div> { /* TODO: include this right here as a table */ }
```
Comment (benj-zen) [inline frontend/src/components/views.js:1144]: See comment above.

File: frontend/src/components/views.js (hunk 1 above)
Comment (benj-zen) [reply frontend/src/components/views.js:881]: Maybe you can add a wrapper analogous to `linkify` for internal links.

**Comment by dingens** [comment]
Fixed your remarks. I've fucked up a commit plus I had multiple master merge commits, so I just force-pushed it. Hope you can cope with that :)

**Review by benj-zen — APPROVED** [review]

**Pull request merged**